File: integrations/vercel-ai-sdk/blocksize-tool.ts
import { tool } from "ai";
import { z } from "zod";

import { getBlocksizeBidAsk, getBlocksizeVwap } from "../typescript/blocksize.js";

export function createBlocksizeTools(agentId: string) {
  return {
    blocksizeVwap: tool({
      description: "Get a live Blocksize VWAP with source timestamp and citation metadata.",
      inputSchema: z.object({ pair: z.string().describe("Instrument such as BTC-USD") }),
      execute: async ({ pair }) => getBlocksizeVwap(pair, agentId),
    }),
    blocksizeBidAsk: tool({
      description: "Get live Blocksize bid/ask data for a supported crypto or equity instrument.",
      inputSchema: z.object({ pair: z.string().describe("Instrument such as ETH-USD or AAPL") }),
      execute: async ({ pair }) => getBlocksizeBidAsk(pair, agentId),
    }),
  };
}

File: integrations/openai_agents/blocksize_tool.py
"""OpenAI Agents SDK function tools for Blocksize production data."""

from __future__ import annotations

import os

from agents import function_tool

from integrations.python import BlocksizeHTTPClient


client = BlocksizeHTTPClient(os.environ.get("BLOCKSIZE_AGENT_ID", "openai-agents-blocksize"))


@function_tool
def blocksize_vwap(pair: str) -> dict:
    """Get a live Blocksize VWAP with its source timestamp and citation metadata.

    Args:
        pair: Instrument such as BTC-USD.
    """
    return client.get_vwap(pair)


@function_tool
def blocksize_bid_ask(pair: str) -> dict:
    """Get a live Blocksize bid/ask for a supported crypto or equity instrument.

    Args:
        pair: Instrument such as ETH-USD or AAPL.
    """
    return client.get_bid_ask(pair)


BLOCKSIZE_TOOLS = [blocksize_vwap, blocksize_bid_ask]
